### PR TITLE
Migrate to scala 2.13

### PR DESCRIPTION
**Pull request by michel-steuwer**
This PR updates Scala to 2.13 and builds upon https://github.com/rise-lang/shine/pull/66 (which should be merged first).

This is marked as a draft PR until #66 is merged.

### CODE REVIEW

**Review by fedepiz — DISMISSED** [review]
Looks good to me!
Thanks for cleaning up so many little things as well!

File: src/main/scala/exploration/riseExploration.scala [hunk 1]
```diff
@@ -26,18 +28,20 @@ object riseExploration {
...
     // code here
 
-    val startingPoint = prepareExploration(parsedConfiguration, solution, filePath)
+    val startingPoint = prepareExploration(parsedConfiguration,
+      solution, filePath)
 
     // start
-    startingPoint.execute(new Solution[Rise](solution, scala.collection.mutable.Seq.empty[Strategy[Rise]]))
+    startingPoint.execute(Solution(solution,
+      immutable.Seq.empty[Strategy[Rise]]))
```
Comment (Bastacyclop) [inline src/main/scala/exploration/riseExploration.scala:44]: I assume changing to an immutable collection is desired?

File: src/main/scala/shine/DPIA/FunctionalPrimitives/AbstractScan.scala [hunk 2]
```diff
@@ -88,7 +88,7 @@ abstract  class AbstractScan(n: Nat,
       </input>
     </reduce>.copy(label = {
       val name = this.getClass.getSimpleName
-      Character.toLowerCase(name.charAt(0)) + name.substring(1)
+      s"${Character.toLowerCase(name.charAt(0))}${name.substring(1)}"
     })
```
Comment (Bastacyclop) [inline src/main/scala/shine/DPIA/FunctionalPrimitives/AbstractScan.scala:92]: maybe we can open an issue to remove the XML printing altogether?

File: src/main/scala/shine/DPIA/MutableIdentityHashMap.scala [hunk 3]
```diff
@@ -1,50 +1,50 @@
...
+//    private val mapIter = innerMap.entrySet.iterator
+//
+//    override def hasNext: Boolean = mapIter.hasNext
+//
+//    override def next(): (K, V) = {
+//      val nextEntry = mapIter.next()
+//      (nextEntry.getKey, nextEntry.getValue)
+//    }
+//  }
+//}
```
Comment (Bastacyclop) [inline src/main/scala/shine/DPIA/MutableIdentityHashMap.scala:50]: is this not required anymore? should it be properly deleted?

File: src/main/scala/shine/DPIA/InferAccessAnnotation.scala [hunk 4]
```diff
@@ -16,17 +16,23 @@ import scala.collection.mutable
 import scala.util.{Failure, Success, Try}
 
 object inferAccess {
-  def apply(e: r.Expr): MutableIdentityHashMap[r.Expr, PhraseType] =
+  def apply(e: r.Expr): java.util.IdentityHashMap[r.Expr, PhraseType] =
+  //MutableIdentityHashMap[r.Expr, PhraseType] =
```
Comment (Bastacyclop) [inline src/main/scala/shine/DPIA/InferAccessAnnotation.scala:20]: should we keep that?

File: src/main/scala/shine/C/AST/Printer.scala [hunk 5]
```diff
@@ -88,7 +88,7 @@ class CPrinter extends Printer {
     case b: Block => printBlock(b)
     case f: ForLoop => printForLoop(f)
     case w: WhileLoop => printWhileLoop(w)
-    case i: IfThenElse => printIfThenElse(i)
+    case i: shine.C.AST.IfThenElse => printIfThenElse(i)
```
Comment (Bastacyclop) [inline src/main/scala/shine/C/AST/Printer.scala:91]: why is this case different now?

File: src/main/scala/exploration/riseExploration.scala (hunk 1 above)
Comment (michel-steuwer) [reply src/main/scala/exploration/riseExploration.scala:44]: Yes. I don't see why the collection was immutable, to begin with.

File: src/main/scala/shine/DPIA/FunctionalPrimitives/AbstractScan.scala (hunk 2 above)
Comment (michel-steuwer) [reply src/main/scala/shine/DPIA/FunctionalPrimitives/AbstractScan.scala:92]: Yes, obviously orthogonal, but I agree: https://github.com/rise-lang/shine/issues/69

File: src/main/scala/shine/DPIA/MutableIdentityHashMap.scala (hunk 3 above)
Comment (michel-steuwer) [reply src/main/scala/shine/DPIA/MutableIdentityHashMap.scala:50]: Yes, good point. I will remove the file.

File: src/main/scala/shine/DPIA/InferAccessAnnotation.scala (hunk 4 above)
Comment (michel-steuwer) [reply src/main/scala/shine/DPIA/InferAccessAnnotation.scala:20]: Nope. I will clean this up as well. Thanks for spotting this.

File: src/main/scala/shine/C/AST/Printer.scala (hunk 5 above)
Comment (michel-steuwer) [reply src/main/scala/shine/C/AST/Printer.scala:91]: I was surprised by this as well. It seems as if the Scala compiler has changed some of its resolution mechanism and was now selecting the wrong `IfThenElse` class.

**Review by Bastacyclop — APPROVED** [review]
:+1:

**Pull request merged**